app/views/programming_groups/index.html.slim

```diff
@@ -0,0 +1,34 @@
+h1 = ProgrammingGroup.model_name.human(count: 2)
+
+- if params[:exercise_id].nil?
+  = render(layout: 'shared/form_filters') do |f|
+    .col-auto
+      = f.label(:exercise_id_eq, t('activerecord.attributes.programming_group.exercise'), class: 'visually-hidden form-label')
+      = f.collection_select(:exercise_id_eq, Exercise.with_programming_groups, :id, :title, class: 'form-control', prompt: t('activerecord.attributes.programming_group.exercise'))
+
+.table-responsive
+  table.table.mt-4 class="#{@programming_groups.present? ? 'sortable' : ''}"
+    thead
+      tr
+        th = sort_link(@search, :id, t('activerecord.attributes.programming_group.programming_group_name'))
+        th = sort_link(@search, :exercise_id, t('activerecord.attributes.programming_group.exercise'))
+        th = sort_link(@search, :user_id, t('activerecord.attributes.programming_group.member'))
+        th = sort_link(@search, :count, t('activerecord.attributes.programming_group.member_count'))
+        th = sort_link(@search, :created_at, t('shared.created_at'))
+        th colspan=3 = t('shared.actions')
+    tbody
+      - if params[:exercise_id].nil?
+        - filtered_programming_groups = @programming_groups
+      - else
+        - filtered_programming_groups = @programming_groups.where(exercise_id: params[:exercise_id])
+      - filtered_programming_groups.each do |programming_group|
+        tr
+          td = link_to_if(policy(programming_group).show?, programming_group.displayname, programming_group)
+          td = link_to_if(policy(programming_group.exercise).show?, programming_group.exercise.title, programming_group.exercise, 'data-turbolinks' => "false")
```

Comment:
Since we might render this page in the context of a specific exercise (`params[:exercise_id]` is set), we should remove this column there as well.